tests/db-updates/class-wc-tests-update-is-read-from-last-read.php

```diff
@@ -0,0 +1,129 @@
+<?php
+
+use Automattic\WooCommerce\Admin\Notes\Note;
+
+/**
+ * DB Update test for wc_admin_update_300_update_is_read_from_last_read()
+ *
+ * @package WooCommerce\Admin\Tests\DBUpdates
+ */
+class WC_Tests_Update_Is_Read_From_Last_Read extends WC_Unit_Test_Case {
+	/**
+	 * @var object current user
+	 */
+	private $user;
+
+	/**
+	 * setUp
+	 */
+	public function setUp() {
+		parent::setUp();
+
+		$this->add_is_read_col();
+		$this->user = $this->factory->user->create(
+			array(
+				'role' => 'administrator',
+			)
+		);
+	}
+
+	/**
+	 * Adds is_read col to the notes table.
+	 */
+	protected function add_is_read_col() {
+		global $wpdb;
+		// make sure is_read col has been added successfully.
+		$notes_table_cols = $wpdb->get_results( "show columns from {$wpdb->prefix}wc_admin_notes" );
```

Comment:
There seems to be some duplication between the code here and the update function; I wonder if we could combine them via a shared method in the notes class.

But really, I'm curious why we're doing so much database checking on columns.  Do we anticipate failure with `dbDelta` or has some occurred during testing?

Reply:
One thing I noticed is that `create_tables` routine doesn't run during the test. Is there a way to run the update routine by any chance?

Reply:
> There seems to be some duplication between the code here and the update function

Do you mean the count query? We could add `get_notes_by($field, $value);` method in the notes class. What do you think?

Reply:
I was more referring to `has_is_read_col` logic.  If we're really set on adding it, we could probably extract that logic into some shared method like `has_is_read_column_exists()`.  Though again, I'd prefer to not have this logic if at all possible.

Reply:
>One thing I noticed is that create_tables routine doesn't run during the test. Is there a way to run the update routine by any chance?

I believe they're run here - https://github.com/woocommerce/woocommerce-admin/blob/80873f1b6730e78dcab23325cc53a930ad68111d/tests/bootstrap.php#L115

It's possible your env is cached and needs to be removed prior to rerunning tests.

Reply:
@joshuatf Thank you for the reference. That's interesting 🤔  I thought I had to run it by myself. I'll look into it 👍